### DISCUSSION TITLE
v3 and downstream testing

### DISCUSSION THREAD

**Discussion by braingram** [zarr-developers/zarr-python#1696] [Q&A] [answered]
Hi!

I maintain a python package that uses zarr:
https://github.com/asdf-format/asdf-zarr

It includes testing of the zarr `main` branch as part of the CI. However I've noticed that the version 3 development is occurring on the `v3` branch (so the above package is not being tested against `v3`).

Would you recommend adding testing against `v3` or will there be a period of time where `v3` will be part of `main` before version 3.0.0 is release?

I read through https://github.com/zarr-developers/zarr-python/blob/main/v3-roadmap-and-design.md#development-process (thanks for putting that together it was very helpful). I wasn't sure of `When v3 is complete, move contents of v3 to the package root` meant that `v3` would replace `main` before or after 3.0.0 was released.

Thanks!

**Comment by jhamman** [comment, marked as answer]
Hi @braingram! Thanks for asking this question and glad you found the v3 design doc helpful. I don't think we're quite to the point where it makes sense for you to start testing against the v3 branch. We're still working on the top level API and we expect a few internal APIs to still change significantly. 

Additionally, we have not discussed the exact plan for how we will release v3 but I do expect it to land in main for a period of time before the final release.

**Reply by braingram** [reply]
Thanks! That is great to know. I'm looking forward to v3.
